src/graph_transpiler/webdnn/frontend/keras/layers/convolutional.py

```diff
@@ -0,0 +1,152 @@
+        y, = AxiswiseBias(None, Axis.C)(y, b)
+
+    y = do_activation(k_op.activation, y)
+    converter.set_variable(converter.get_output_tensor(k_op)[0], y)
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("SeparableConv2D")
+def _convert_separable_conv2d(converter: KerasConverter, k_op: keras.layers.SeparableConv2D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.SeparableConv2D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("Conv2DTranspose")
+def _convert_conv2d_transpose(converter: KerasConverter, k_op: keras.layers.Conv2DTranspose):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.Conv2DTranspose is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("Conv3D")
+def _convert_conv3d(converter: KerasConverter, k_op: keras.layers.Conv3D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.Conv3D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("Cropping1D")
+def _convert_cropping1d(converter: KerasConverter, k_op: keras.layers.Cropping1D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.Cropping1D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("Cropping2D")
+def _convert_cropping2d(converter: KerasConverter, k_op: keras.layers.Cropping2D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.Cropping2D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("Cropping3D")
+def _convert_cropping3d(converter: KerasConverter, k_op: keras.layers.Cropping3D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.Cropping3D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("UpSampling1D")
+def _convert_up_sampling1d(converter: KerasConverter, k_op: keras.layers.UpSampling1D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.UpSampling1D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("UpSampling2D")
+def _convert_up_sampling2d(converter: KerasConverter, k_op: keras.layers.UpSampling2D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.UpSampling2D is not supported')
+
+
+# noinspection PyUnusedLocal
+@KerasConverter.register_handler("UpSampling3D")
+def _convert_up_sampling2d(converter: KerasConverter, k_op: keras.layers.UpSampling3D):
+    # TODO
+    raise NotImplementedError('[KerasConverter] keras.layers.UpSampling3D is not supported')
+
+
+@KerasConverter.register_handler("ZeroPadding1D")
+def _convert_zero_padding1d(converter: KerasConverter, k_op: keras.layers.ZeroPadding1D):
+    x = converter.get_variable(converter.get_input_tensor(k_op)[0])
+
+    assert x.order == OrderNTC
+    if k_op.padding[0][0] != k_op.padding[0][1]:
```

Comment:
This is not necessarily the same.
Padding2D only supports when top==bottom and left==right, but this restriction should be removed.